hangman.py

```diff
@@ -0,0 +1,330 @@
+            guesses_left = results(guessed_letter, guesses_left, letters_in_secret_word, secret_word, letters_guessed)
+        if is_word_guessed(secret_word, letters_guessed):
+            return True
+
+def hangman(secret_word, hints_switch = False):
+    '''
+    secret_word: string, the secret word to guess.
+
+    Starts up an interactive game of Hangman.
+
+    * At the start of the game, let the user know how many
+      letters the secret_word contains and how many guesses s/he starts with.
+
+    * The user should start with 6 guesses
+
+    * Before each round, you should display to the user how many guesses
+      s/he has left and the letters that the user has not yet guessed.
+
+    * Ask the user to supply one guess per round. Remember to make
+      sure that the user puts in a letter!
+
+    * The user should receive feedback immediately after each guess
+      about whether their guess appears in the computer's word.
+
+    * After each guess, you should display to the user the
+      partially guessed word so far.
+
+    Follows the other limitations detailed in the problem write-up.
+    '''
+    #Instruction to the game for player
+    letters_in_secret_word = set(secret_word)
+    print("Welcome to the game Hangman!")
+    print("I am thinking of a word that is", len(secret_word), "letters long.")
+    print("You have 3 warnings left")
+    guesses_left = GUESSES_AT_START
+    warnings_left = WARNINGS_AT_START
+    letters_guessed = []
+    winning = logical_body_hangman(guesses_left, letters_guessed, warnings_left, letters_in_secret_word, hints_switch)
+    end_of_the_game(winning, guesses_left, secret_word)
+
+
+def match_with_gaps(my_word, other_word, my_word_for_dublicate):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    other_word: string, regular English word
+    returns: boolean, True if all the actual letters of my_word match the
+        corresponding letters of other_word, or the letter is the special symbol
+        _ , and my_word and other_word are of the same length;
+        False otherwise:
+    '''
+    #Two lists above are needed to delete  space gaps in my_word
+    #Then they are used to compare lengths of my_word without space gaps and then compare every symbols
+    if len(my_word) != len(other_word):
+        return False
+    for symbol in range(0, len(my_word)):
+        if my_word[symbol] == '_' or my_word[symbol] == other_word[symbol]:
+            continue
+        return False
+   #Code below is needed for not showing words, if they have same letters and length,
+   #but my_word has any letter and other_word has two same letters as letter in my_word
+   #For example, function doesn't show word 'apple' if my_word is 'a_ ple'
+    for symbol in range(0, len(my_word_for_dublicate)):
+        #Counting number of every letter in both words and comparing them
+        letters_1 = my_word_for_dublicate.count(my_word_for_dublicate[symbol])
```

Comment:
Effectively - it's not a letter. It's a number of letter occurrence. And try to avoid such number in names.

Reply:
done